flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointITCase.java

```diff
@@ -531,8 +534,10 @@ public void invoke(Long value, Context context) throws Exception {
 			state.numOutput++;
 
 			if (state.completedCheckpoints < minCheckpoints) {
-				// induce heavy backpressure until enough checkpoints have been written
-				Thread.sleep(0, 100_000);
+				// induce backpressure until enough checkpoints have been written
+				if (random.nextInt(1000) == 42) {
```

Comment:
Why do we need to reduce the backpressure here? I'm worried that any solution on top of it will only work with a certain minimal flow.

Note that in https://github.com/apache/flink/pull/13845/commits/ef4f78d9b7bedb10a8c7be3e062032d2e7a0c77c#diff-c0d448d637f04cc203de3592d7fab0ba2412f2d75c59dcf6e44cc4a3c18e5851R568 I added a modification to the test that avoids backpressuring during recovery (by applying backpressure only after the first successful checkpoint.

Reply:
> Why do we need to reduce the backpressure here?

This is a commit from #13827. As far as I understand, the motivation was:
1. to have higher backpressure (.1ms vs 1ms) for **some** parts of the stream
2. to have unbalanced streams
3. reduce running time (was ~1m, now ~10s)

On average, the change decreases sleep time only by 60% (1-42/1000*10).

> I'm worried that any solution on top of it will only work with a certain minimal flow.

With 1K elements, the sink will sleep 23 times for 1ms, which I think is enough and 1K is a reasonable number of elements for this test.